File: src/elements/City.cpp
/***************************************************************************
                          City.cpp  -  description
                             -------------------
    begin                : Sat May 10 2003
    copyright            : (C) 2001-2003 by Freerails developers
    email                :
 ***************************************************************************/

/***************************************************************************
 *                                                                         *
 *   This program is free software; you can redistribute it and/or modify  *
 *   it under the terms of the GNU General Public License as published by  *
 *   the Free Software Foundation; either version 2 of the License, or     *
 *   (at your option) any later version.                                   *
 *                                                                         *
 ***************************************************************************/

#include "City.h"

City::City(unsigned int _posX, unsigned int _posY, std::string _name)
       : GamePosElement(_posX, _posY, NULL, idCity)
{
  name = _name;
}

City::~City()
{
}

void City::serialize(Serializer* _serializer)
{
  GamePosElement::serialize(_serializer);
  *_serializer << (const std::string)name;
}

void City::deserialize(Serializer* _serializer)
{
  GamePosElement::deserialize(_serializer);
  *_serializer >> name;
}

File: src/main/java/freerails/model/terrain/City.java
/*
 * FreeRails
 * Copyright (C) 2000-2018 The FreeRails Team
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */

package freerails.model.terrain;

import freerails.model.Identifiable;
import freerails.util.Vec2D;

// TODO Possible potential for expansion?? Initial size of city, growth rate etc.??? incorporate CityModel?
/**
 * Simply stores the name and x and y coordinates of a city.
 */
public class City extends Identifiable {

    private final String name;
    private final Vec2D location;

    public City(int id, String name, Vec2D location) {
        super(id);
        this.name = name;
        this.location = location;
    }

    public String getName() {
        return name;
    }

    public Vec2D getLocation() {
        return location;
    }

    @Override
    public String toString() {
        return String.format("%s %s", name, location);
    }
}
